### ISSUE TITLE
Optimise the config files handling

### ISSUE THREAD

**Issue by YipengHu** [DeepRegNet/DeepReg#634]
## Subject of the feature

Reading and processing config file can be improved:
- For those fields that are not required for individual options, warning/error should not be raised (tba): 
*global net does not need extract_levels or method (predicting dvf or conditional makes no difference to ddf, even enabled)
*train does not require test (or even valid) dataset(s)
*unet does not require extract_levels

- The section `dataset` and `train` may be regrouped due to the requirement for train and predict, consider:
*`dataset`: required for train and predict, but with different requirement
**train requires train, train/val, train/test or train/val/test (this can be complicated, so *minimally require no test*)
**predict requires only test dataset
*`model`: required for both train and test and have to be consistent between the two, including current `method` and `backbone`
*`train`: hyperparameters only required during training, and not required for predict, including current `loss`, `optimizer`, `preprocess`, `epoch`, `save_period`

- Disable `label_sample` for predict, while predict will use all labels provided.

If the feature request is approved, would you be willing to submit a PR? _(Help can be
provided if you need assistance submitting a PR)_

Yes

**Comment by NMontanaBrown** [comment]
Watch out for backcompatibility and modifs here

**Comment by mathpluscode** [comment]
> Watch out for backcompatibility and modifs here

I got an idea for back-compatibility.

As we are doing config checks, we can add one function, to enable old -> latest config transformation automatically and print warnings to remind users! As a result, code will be back-compatible! @YipengHu @NMontanaBrown

**Comment by YipengHu** [comment]
@mathpluscode why is this not in Milestone?

**Comment by NMontanaBrown** [comment]
What's the status check for this? @YipengHu

**Comment by mathpluscode** [comment]
ok, will work on it ^^
